lib/rubocop/ast/node.rb

```diff
@@ -101,11 +105,19 @@ def initialize(type, children = [], properties = {})
         end
       end
 
-      Parser::Meta::NODE_TYPES.each do |node_type|
+      (Parser::Meta::NODE_TYPES - [:send]).each do |node_type|
         method_name = "#{node_type.to_s.gsub(/\W/, '')}_type?"
-        define_method(method_name) do
-          type == node_type
-        end
+        class_eval <<~RUBY, __FILE__, __LINE__ + 1
+          def #{method_name}          # def block_type?
+            @type == :#{node_type}    #   @type == :block
+          end                         # end
+        RUBY
+      end
+
+      # Most nodes are of 'send' type, so this method is defined
+      # separately to make this check as fast as possible.
+      def send_type?
+        false
       end
```

Comment:
Most checks are done against `send` nodes, so I extracted a dedicated fastest method for this.